src/java/com/unifina/signalpath/ModuleWithUI.java

```diff
@@ -72,17 +101,30 @@ private StreamService getStreamService() {
 		return streamService;
 	}
 
-	public void pushToUiChannel(Map content) {
-		Stream stream = getUiChannel().getStream();
+	private StreamMessage getMsg(Stream stream, Map content){
 		int streamPartition = StreamPartitioner.partition(stream, null);
+		long timestamp = System.currentTimeMillis();
+		long sequenceNumber = getNextSequenceNumber(stream.getId(), timestamp);
+		SecUser user = SecUser.getViaJava(getGlobals().getUserId());
```

Comment:
Can't do a database lookup for every event, too slow! The user can't change during runtime, so you can cache it into a variable and reuse it. For safe serialization/deserialization, you might wanna make it `transient` and re-fetch it if it's null:

```
private transient SecUser cachedUser = null;

private SecUser getUser() {
    if (cachedUser == null) {
        cachedUser = SecUser.getViaJava(getGlobals().getUserId());
    }
    return cachedUser;
}

    // ... then in getMsg
    SecUser user = getUser();
```